### ISSUE TITLE
Datatype mismatch exception in case of argument binding with LIMIT keyword

### ISSUE THREAD

**Issue by pPanda-beta** [aminroosta/sqlite_modern_cpp#123]
for queries with LIMIT keyword its giving "datatype_mismatch" exception
_But works and gives correct error in case we provide a value higher than maximum row index. 
So that clearly says there is no problem with datatype._

To reproduce :
```
	try
	{
		database db(":memory:");
		db << "CREATE TABLE tbl (id integer, name string);";
		db << "INSERT INTO tbl VALUES (?, ?);" << 10 << "hello";
		db << "INSERT INTO tbl VALUES (?, ?);" << 20 << "world";
		int id = -1;
		//Works fine
		db<<"SELECT id FROM tbl LIMIT 0,1;">>id;
		cout<<id<<"\n";
		//Worst experience ever!
		db<<"SELECT id FROM tbl LIMIT ?,1;"<< 0 >>id;
		cout<<id<<"\n";
	}
	catch(sqlite_exception &e)
	{
		cerr<<e.what()<<"\n"
		    <<e.get_code()<<"\n"
		    <<e.get_sql()<<"\n";
	}
```

Output : 
'''
10
datatype mismatch
20
SELECT id FROM tbl LIMIT NULL,1;
'''

**Comment by zauguin** [comment]
Which platform and versions do you use?
On linux with clang or gcc and the current master I can't reproduce it right now.
I get your output if I instead write 

    db<<"SELECT id FROM tbl LIMIT ?,1;" >>id;

But that's expected.

**Comment by zauguin** [comment]
@pPanda-beta The test I used: https://github.com/zauguin/sqlite_modern_cpp/blob/bugfix/tests/bound_limit.cc

**Comment by pPanda-beta** [comment]
Sorry my bad!
This bug isnt happening with latest build. But its happening with release 3.1. 
I've double checked since currently in a project I'm forced to use release 3.1 instead of 3.2 or any further as its troubling. 
I think it has already been fixed in 3.2 but with a different issue.

**Comment by pPanda-beta** [comment]
I guess I can close it.

**Comment by zauguin** [comment]
Right, it's fixed with 2df66f1147864a62bfa19f358dfe02b766830d29. Maybe you can apply just this patch, it should work isolated.
